mica/stats/mag_stats/mag_stats.py

```diff
@@ -0,0 +1,790 @@
+                result = func(*args, **kwargs)
+                result.write(filename)
+                return result
+        return wrapper
+    return decorator_cache
+
+
+#@astropy_table_cache(name='telem', dir='/Users/javierg/SAO/mica/mag_stats_cache/telem')
+def get_telemetry_by_agasc_id(agasc_id, obsid=None, ignore_exceptions=False):
+    """
+    Get all telemetry relevant for the mag_stats task for a given AGASC ID.
+
+    :param agasc_id: int
+    :param obsid: int (optional)
+    :return: dict
+    """
+    catalogs.load()
+    if obsid is None:
+        obs = catalogs.STARS_OBS[
+            (catalogs.STARS_OBS['agasc_id'] == agasc_id)]
+    else:
+        obs = catalogs.STARS_OBS[(catalogs.STARS_OBS['agasc_id'] == agasc_id) &
+                                 (catalogs.STARS_OBS['obsid'] == obsid)]
+    if len(obs) > 1:
+        obs = obs.loc['mp_starcat_time', sorted(obs['mp_starcat_time'])]
+    telem = []
+    for i, o in enumerate(obs):
+        try:
+            t = Table(get_telemetry(o))
+            t['obsid'] = o['obsid']
+            t['agasc_id'] = agasc_id
+            telem.append(t)
+        except Exception as e:
+            import sys
+            import traceback
+            print(f'{agasc_id} failed', e)
+            exc_type, exc_value, exc_traceback = sys.exc_info()
+            trace = traceback.extract_tb(exc_traceback)
+            print(f'{exc_type.__name__} {exc_value}')
+            for step in trace:
+                print(f'  in {step.filename}:{step.lineno}/{step.name}:')
+                print(f'    {step.line}')
+            if not ignore_exceptions:
+                raise
+    return vstack(telem)
+
+
+def add_obsid_info(telem, obs_stats):
+    obs_stats['obsid_ok'] = (
+        (obs_stats['n'] > 10) &
+        (obs_stats['f_ok'] > 0.3) &
+        (obs_stats['lf_variability_100s'] < 1)
+    )
+    obs_stats['comments'] = np.zeros(len(obs_stats), dtype='<U80')
+
+    telem = vstack(telem)
+    telem['obsid_ok'] = True
+    telem['obsid_outlier'] = False
+
+    for s in obs_stats:
+        obsid = s['obsid']
+        o = (telem['obsid'] == obsid)
+        telem['obsid_ok'][o] = np.ones(np.sum(o), dtype=bool) * s['obsid_ok']
+        if (np.any(telem['ok'][o]) and s['f_track'] > 0 and
+                np.isfinite(s['q75']) and np.isfinite(s['q25'])):
+            iqr = s['q75'] - s['q25']
+            telem['obsid_outlier'][o] = (telem[o]['ok'] & (iqr > 0) &
+                                         (telem[o]['mags'] < s['q25'] - 1.5*iqr) |
+                                         (telem[o]['mags'] > s['q75'] + 1.5*iqr))
+    return telem
+
+
+@numba.jit(nopython=True)
+def staggered_aca_slice(array_in, array_out, row, col):
+    for i in np.arange(len(row)):
+        if row[i]+8 < 1024 and col[i]+8 < 1024:
+            array_out[i] = array_in[row[i]:row[i]+8, col[i]:col[i]+8]
+
+
+def get_mag_from_img(slot_data, t_start, ok=True):
+    dark_cal = get_dark_cal_image(t_start, 'nearest',
```

Comment:
Was there a notebook that justified removing the dark cal?  If so, link here would be good.

Reply:
Or wait, is this subtracting the dark cal or just saving it for later?

Reply:
And do we need *some* kind of background subtraction?